src/main/java/com/example/airbnbapi/api/Controller.java

```diff
@@ -18,25 +17,39 @@
 
 
     @Autowired
-    public Controller( ServiceFactory serviceFactory) {
+    public Controller(ServiceFactory serviceFactory) {
 
         this.serviceFactory = serviceFactory;
 
     }
 
     @GetMapping(path = "/{type}")
-    public  Media[] getAllMedia (@PathVariable("type") MediaType type) {
-      return   serviceFactory.getServiceByType(type).getItems();
+    public ResponseEntity<Media[]> getAllMedia(@PathVariable("type") MediaType type) {
+
+        Media[] items = serviceFactory.getServiceByType(type).getItems();
+        if (items != null) {
+            return new ResponseEntity<>(items, HttpStatus.OK);
+
+        } else {
+            return new ResponseEntity<Media[]>(HttpStatus.BAD_REQUEST);
+
+        }
 
     }
 
     @GetMapping(path = "/{type}/{id}")
 
-    public Media getGameById (@PathVariable("type") MediaType type, @PathVariable("id") int id){
+    public ResponseEntity<Media> getGameById(@PathVariable("type") MediaType type, @PathVariable("id") int id) {
+
+       Media item = serviceFactory.getServiceByType(type).getMediaById(id);
 
+        if (item != null) {
+            return new ResponseEntity<Media>(item, HttpStatus.OK);
 
+        } else {
+            return new ResponseEntity<Media>(HttpStatus.BAD_REQUEST);
```

Comment:
Consider using `NOT_FOUND` - `BAD_REQUEST`'s are typically reserved for client submissions